lib/modules/versioning/nuget/index.ts

```diff
@@ -1,36 +1,260 @@
-import semver from 'semver';
-import { regEx } from '../../../util/regex';
-import { GenericVersion, GenericVersioningApi } from '../generic';
-import type { VersioningApi } from '../types';
+import type { NewValueConfig, VersioningApi } from '../types';
+import { cmp } from './compare';
+import { parseRange, parseVersion } from './parse';
+import * as nugetRange from './range';
+import type { NugetVersion } from './types';
 
 export const id = 'nuget';
 export const displayName = 'NuGet';
 export const urls = [
   'https://docs.microsoft.com/en-us/nuget/concepts/package-versioning',
 ];
-export const supportsRanges = false;
+export const supportsRanges = true;
+export const supportedRangeStrategies = ['pin', 'replace'];
 
-const pattern = regEx(/^(\d+(?:\.\d+)*)(-[^+]+)?(\+.*)?$/);
+class NugetVersioningApi implements VersioningApi {
+  isCompatible(version: string, _current?: string | undefined): boolean {
+    return this.isValid(version);
+  }
+
+  isSingleVersion(version: string): boolean {
+    if (this.isVersion(version)) {
+      return true;
+    }
```

Comment:
did nuget versioning have isSingleVersion before? or it just defaulted to "isVersion"?

Reply:
defaulted to isVersion in base class